msteams-platform/m365-apps/specify-runtime-requirements.md

```diff
@@ -0,0 +1,132 @@
+- Bots (`bots`)
+- Configurable tabs (`configurableTabs`)
+
+Each app element is denoted by an `id`, which maps to `botId` for bots, `entityId` for staticTabs, `id` for configurableTabs, and `id` for composeExtensions.
+
+> [!NOTE]
+> This feature introduces `id` properties to configurableTabs and composeExtensions definitions. Currently, Teams and other Microsoft 365 hosts only support apps containing a single configurableTab and/or single composeExtension. These ID properties were added to future-proof your app in case host support expands to accommodate multiple instances of these components.
+>
+> The `id` property must be specified on a configurableTab or composeExtension for it to be recognized in an `elementRelationshipSet`.
+
+### One-way dependencies
+
+Use the [`oneWayDependencies`](../resources/schema/manifest-schema-dev-preview.md#elementrelationshipsetonewaydependency) array to describe cases where one component of your app depends upon another component. For each object in the array, specify the dependent component (`element`) and the component it depends on (`dependsOn`). For example:
+
+```json
+    "elementRelationshipSet": {
+      "oneWayDependencies" : [
+        {
+          "element" : {
+            "name" : "composeExtensions",
+            "id" : "composeExtension-id",
+            "commandIds": ["command-1-id", "command-2-id"]  // Developers can add more commands.
+          },
+          "dependsOn" : [
+              {"name" : "bots", "id" : "bot-id"}
+            ]
+        }
+      ]
+    }
+```
+
+For message extensions, you can optionally specify individual commands that require support for specific app components. If those components aren't supported in the runtime host, they will not be made available to the end-user (though all other commands will run).
+
+### Mutual dependencies
+
+Use the [`mutualDependencies`](../resources/schema/manifest-schema-dev-preview.md#elementrelationshipsetmutualdependencies) array to group app components that must load together in order to support their intended function. Each object in the array represents a mutually dependent app component. For example:
+
+```json
+    "elementRelationshipSet": {
+      "mutualDependencies" : [
+                {"name" : "bots", "id" : "bot-id"}, 
+                {"name" : "staticTabs", "id" : "staticTab-id"},
+                {"name" : "composeExtensions", "id" : "composeExtension-id"},
+                {"name" : "configurableTabs", "id": "configurableTab-id"}
+      ]
+    },
+```
+
+## Specify runtime capability requirements for specific app components (`requirementSet`)
+
+When you're defining an individual app component, you can specify it's specific TeamsJS runtime requirements using a [`requirementSet`](../resources/schema/manifest-schema-dev-preview.md#statictabsrequirementset). This will ensure the component only loads in Microsoft 365 hosts with support for the critical TeamsJS capabilities. For example:
```

Comment:
```suggestion
When you're defining an individual app component, you can specify its specific TeamsJS runtime requirements using a [`requirementSet`](../resources/schema/manifest-schema-dev-preview.md#statictabsrequirementset). This will ensure the component only loads in Microsoft 365 hosts with support for the critical TeamsJS capabilities. For example:
```